From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Tue, 9 Jan 2024 05:58:19 +0000
Subject: [PATCH] Extension of build system for java files in base layer

Adapted from GrapheneOS Vanadium patch 0142-Extension-of-build-system-for-java-files-in-base-lay.patch for the Vandium browser port.

---
 base/BUILD.gn                     | 10 ++++++++++
 base/base_android_library_ext.gni | 21 +++++++++++++++++++++
 2 files changed, 31 insertions(+)
 create mode 100644 base/base_android_library_ext.gni

diff --git a/base/BUILD.gn b/base/BUILD.gn
index 695316a0d5066..5ce00764f8a3a 100644
--- a/base/BUILD.gn
+++ b/base/BUILD.gn
@@ -4687,6 +4687,11 @@ if (is_android) {
       ":wrapped_class_loader_java",
     ]
 
+    import("base_android_library_ext.gni")
+    sources += base_java_ext_java_sources
+    deps += base_java_ext_deps
+    srcjar_deps += base_java_ext_srcjar_deps
+
     if (!is_cronet_build) {
       proguard_configs += [
         "android/proguard/chromium_apk.flags",
@@ -5293,6 +5298,11 @@ if (is_android) {
       "//third_party/androidx:androidx_annotation_annotation_java",
       "//third_party/jni_zero:jni_zero_java",
     ]
+
+    import("base_android_library_ext.gni")
+    sources += base_library_loader_java_ext_java_sources
+    deps += base_library_loader_java_ext_deps
+    srcjar_deps += base_library_loader_java_ext_srcjar_deps
   }
 
   generate_jni("base_shared_preferences_jni") {
